Update taint-mode docs (#488)
returntocorp/semgrep-docs: docs/writing-rules/data-flow/taint-mode.md

    - pattern: $PATH
```

Metavariables, rule message, and unification
--------------------------------------------

The patterns specified by `pattern-sources` and `pattern-sinks` (and `pattern-sanitizers`) are all independent of each other. If a metavariable used in `pattern-sources` has the same name as a metavariable used in `pattern-sinks`, these are still different metavariables.

> In the message of a taint-mode rule, you can refer to any metavariable bound by `pattern-sinks`, as well as to any metavariable bound by `pattern-sources` that does not conflict with a metavariable bound by `pattern-sinks`.

Author: The rule message is the `message` field of the rule.

> In the message of a taint-mode rule, you can refer to any metavariable bound by `pattern-sinks`, as well as to any metavariable bound by `pattern-sources` that does not conflict with a metavariable bound by `pattern-sinks`.

Reviewer: Maybe we can specify it this way...

Semgrep can treat metavariables with the same name as the _same_ metavariable. This can be done through rule `options` by setting `taint_unify_mvars: true`. Unification enforces that whatever a metavariable binds to in each of these operators is, syntactically speaking, the **same** piece of code. For example, if a metavariable binds to a code variable `x` in the source match, it must bind to the same code variable `x` in the sink match. In general, unless you know what you are doing, avoid metavariable unification between sources and sinks.

:::info
Semgrep used to have a different behavior, see release notes for 0.87.0 for more information.
:::

> For example, here is a rule where unification of source and sink metavariables is useful:

Reviewer: ```suggestion
The following example demonstrates the use of source and sink metavariables.```

Author: We're not demonstrating metavariables, we're demonstrating the unification of metavariables.
